lib/WebSocket.js

```diff
@@ -188,17 +188,13 @@ class WebSocket extends EventEmitter {
     this.emit('close', this._closeCode || 1000, this._closeMessage || '');
 
     if (this._socket) {
-      if (this._ultron) this._ultron.destroy();
+      this._ultron.destroy();
       this._socket.on('error', function onerror () {
-        try {
-          this.destroy();
-        } catch (e) {}
+        this.destroy();
```

Comment:
Calling `socket.destroy` does not seem to throw an error on the same tick in any circumstance.